pandas/tests/series/indexing/test_indexing.py

```diff
@@ -768,3 +768,25 @@ def test_head_tail(test_data):
     assert_series_equal(test_data.series.head(0), test_data.series[0:0])
     assert_series_equal(test_data.series.tail(), test_data.series[-5:])
     assert_series_equal(test_data.series.tail(0), test_data.series[0:0])
+
+
+def test_reset_index_drop_errmsg():
+    # https://github.com/pandas-dev/pandas/issues/20925
+    # Data for Test Case 1 and 2
+    s = pd.Series(range(4))
+    # Test Case 1
+    with tm.assert_raises_regex(KeyError, 'must be same as name'):
+        s.reset_index('wrong', drop=True)
+    # Test Case 2
+    with tm.assert_raises_regex(KeyError, 'must be same as name'):
+        s.reset_index('wrong')
+    # Data for Test Case 3
+    s = pd.Series(range(4), index=pd.MultiIndex.from_product([[1, 2]] * 2))
+    # Test Case 3
+    with tm.assert_raises_regex(KeyError, 'not found'):
+        s.reset_index('wrong', drop=True)
+    # Data for Test Case 4
+    s = pd.Series(range(4), name='valid')
```

Comment:
This (and I mean: my [testcase](https://github.com/pandas-dev/pandas/issues/20925#issuecomment-386284093)) doesn't make much sense, as we are giving a name to ``s`` itself, not to its index. I should have written

```python
s = pd.Series(range(4), index=pd.RangeIndex(4, name='valid'))
```

Notice that if you replace with the above, the following doesn't raise any more. But it might be a bug in ``MultiIndex.droplevel``, in which case OK to postpone.

Reply:
Will delete the Test Case 4 now